include/picongpu/simulationControl/MySimulation.hpp

```diff
@@ -117,6 +117,7 @@ class MySimulation : public SimulationHelper<simDim>
     cellDescription(nullptr),
     initialiserController(nullptr),
     slidingWindow(false),
+    endSlidingOnStep(0),
```

Comment:
this should be `-1` for consistency with the default value. Or you make it `uint32_t` as the usage in `MovingWindow` and put `0` as a default, which I find reasonable a well as a special value (do *not* stop at zero).

Reply:
To enable the moving window we have a bool switch. Therefore I set the default to `disbale/0`
`uint32_t` is not useful because than I need to show the user a magic large number as default in the help (https://github.com/ComputationalRadiationPhysics/picongpu/pull/2792/files). It is more common to have `-1` as unlimited.

Reply:
But in `MovingWindow` you set it to `0`. I would be fine to make it all-`0` as default as you got it there.

Reply:
The object `MovingWindow` is Independent of `MySimulation` there is no need that internal variables of two different objects fits in his default value.
IMO `-1` fits much better to `--stopWindow step` instead `zero`

Reply:
Sure, but following that logic, shouldn't `MovingWindow` also use `-1` for "do not move"? ;-)

Reply:
Also, this has a bug. If a user *would* truly request `--stopWindow 0` then it would still not stop at zero. Therefore, make the cmd line interface with `0 means not stopping` as well for simplicity?

Reply:
update, no wait, it works as intended.